### DISCUSSION TITLE
Function has unresolved stack usage

### DISCUSSION THREAD

**Discussion by ishroudd** [Vector35/binaryninja-api#1644] [General] [answered]
I'm receiving the following warnings on a couple functions:

![Untitled](https://user-images.githubusercontent.com/41129603/81121685-2b65c900-8ee4-11ea-92d5-e3b3ef8718da.png)
![Untitled1](https://user-images.githubusercontent.com/41129603/81121689-2c96f600-8ee4-11ea-97af-6b3450cf9fc8.png)

What exactly does this warning mean, and how would I go about fixing it?

**Comment by plafosse** [comment, marked as answer]
This issue occurs when our analysis of the function can not determine which parameters belong to which function call. In the above we were unable to tell which of the push instructions belong to which call instruction. The way this must be fixed is by right-clicking on the call instruction and specifying what the stack adjustment should be.

**Reply by plafosse** [reply]
If this answers your question please click "mark as answer". Thanks :)

**Reply by houseofwealth** [reply]
I'm sorry but could you elaborate on this a bit. I'm seeing the same warning and its not obvious to me what the problem is exactly or what I'm supposed to adjust on the stack. For example I see a teal box that says:
```
Unresolved call
Stores at offset(s) 0
010016fb call dword [exit@IAT]

```
Also what do the different colors (green and teal) mean? I assume grey means normal). 
thanks

**Reply by bjm9991** [reply]
Hmm, yup it seems this still hasn't been addressed in 2023. I think the GUI solution is to right-click the `call`s and select "Set Stack Adjustment" for each of the unlabelled calls (often only one). However, it seems some of these are standard library calls such as the `basic_stringstream` constructor - so we need more guidance here...

**Reply by xusheng6** [reply]
@commonbreed any chance you can share any of those with me so that we can use that as an example in the documentation? Or we can recreate a sample based on that
